backend/api/measure/auth.go

```diff
@@ -103,6 +116,9 @@ func ValidateAccessToken() gin.HandlerFunc {
 		if claims, ok := accessToken.Claims.(jwt.MapClaims); ok {
 			userId := claims["sub"]
 			c.Set("userId", userId)
+
+			ownTeamId := claims["oti"]
+			c.Set("ownTeamId", ownTeamId)
```

Comment:
@anupcowkur I would strongly argue that we don't set team id in context and we read the user's team information on every request by making an authoritative database read. I don't want to introduce dubious and ambiguous authorization issue cases at the cost of saving 1 database call when it comes to product security. Product security in this case is paramount in my opinion.

Reply:
Not sure how this affects product security. If a user is removed from his default team (which shouldn't happen as per my earlier comment) after a session is created, trying to do any operations on this `own_team_id` would result in authorization failure since the user does not have access anymore. 

The database read would also return an error if the user doesn't have a team anymore. That logic needs to be revisited and we need to make sure that:
1. user always has a team
2. isn't removed from default team

Reply:
as mentioned in another comment, there is a bug which allows user to be removed from their default team. Issue here: #2067 

fixing that issue should alleviate this concern. User will not be able to be removed from their default team making `own_team_id` always valid.

Reply:
I've explained my concerns in the above comment. Additionally, to answer your point:

> Not sure how this affects it.

My point is, the `own_team_id` in the context may be outdated because some other team member changed the role or disabled the user (assume we build something like that in future), hence, I'm saying we should always query the database for the latest state for this kind of information. Think of this not just limited to `own_team_id` but any other variable related to user <> team association, like a banned or disabled user for example. Whichever API call is just relying on reading data from the context without making an explicit db call may become vulnerable to authorization related security issues.

Reply:
Will read from team memberships instead of storing this in auth session